### DISCUSSION TITLE
Why does accessing base automatically add ‘/’ after the route?

### DISCUSSION THREAD

**Discussion by LeeSamFong** [vuejs/router#2220] [Help and Questions] [answered]
### Reproduction

https://github.com/LeeSamFong/vue-router-demo

### Steps to reproduce the bug

1. Visit: http://localhost:5173/dist/aaa
2. Observe that the browser address bar changes to: http://localhost:5173/dist/aaa/
There is an extra `/` at the end
---
`/dist/aaa` -> `/dist/aaa/`

### Expected behavior

When base is not the default value, `/` is not automatically added when accessing the home page.

### Actual behavior

<img width="718" alt="image" src="https://github.com/vuejs/router/assets/47028881/3d2076b0-1b7f-4407-a1fb-4316e9c7083a">
<img width="778" alt="image" src="https://github.com/vuejs/router/assets/47028881/422714da-15c7-4d0f-b870-b709a61bbc8e">

---

When my base is not the default value, after accessing the root route `/dist/aaa`, `/` will be automatically added at the end, becoming ->`/dist/aaa/`

**Comment by posva** [comment, marked as answer]
It’s a convention, similar to example.com becoming example.com/.

**Comment by LeeSamFong** [comment]
But if I don't set the base, `http://localhost:5173` will not be appended with `/`, and it will be automatically removed when '/' is included. Behaving inconsistently.

---

So is this a bug?

**Comment by posva** [comment]
it's not a bug, the browser hides the slash but if you check `location`, it's there
